### PR TITLE
Rewrite /init in Go

### PR DESCRIPTION
**Pull request by justincormack**
This uses a more memory efficient copy, and gets us closer to
not having a shell in the base system if not required.

Signed-off-by: Justin Cormack <justin.cormack@docker.com>

replaces #2098 by using Go not a shell script so we avoid lots of forking. Surprisingly large amount of Go to replace a two line shell script...

walrus

### CODE REVIEW

File: pkg/init/init [hunk 1]
```diff
@@ -1,9 +0,0 @@
-#!/bin/sh
-
-/bin/mount -t tmpfs tmpfs /mnt
-
-/bin/cp -a / /mnt 2>/dev/null
-
-/bin/mount -t proc -o noexec,nosuid,nodev proc /proc
```
Comment (rn) [inline pkg/init/init]: your go code does not mount `/proc` is that only mounted here for the `switch_root`?

Comment (justincormack) [inline pkg/init/init]: Yes, it is not needed, it is done as the first line of the init script in `/bin/rc.init`

**Comment by justincormack** [comment]
Oh, I did update the hash but forgot to push it

**Comment by justincormack** [comment]
Hashes pushed

**Comment by justincormack** [comment]
@rn its green now with updated hashes

**Pull request merged**